### ISSUE TITLE
Removing a notification's icon

### ISSUE THREAD

**Issue by hokomo** [dunst-project/dunst#1010]
### Issue description

I want to be able to remove the icon for all notifications that match a certain pattern.

Setting `new_icon` to the empty string `""` causes Dunst to report `WARNING: Cannot set empty value for setting new_icon` and the icon remains unchanged.

Setting `new_icon` to something like `/dev/null` sort of works, but Dunst reports a warning which is a little bit hacky:

```
WARNING: Failed to load image info for /dev/null                                                                 
WARNING: No icon found in path: '/dev/null'
```
I could get around the warning by using a 1x1 transparent icon for example, but that's a hack as well. :)

Is there a cleaner way of doing this? Could you allow the empty string as a value for `new_icon`?

I've also noticed that the icon override only works with certain notifications. I've opened a separate issue for that (#1009).

### Installation info

- Version: `Dunst - A customizable and lightweight notification-daemon 1.7.3`
- Install type: `package`
- Window manager / Desktop environment: `Fedora 35, i3`

<details>
<summary>Minimal dunstrc</summary>

```ini
[remove_icon]
summary = "*something*"
new_icon = "/dev/null"
```
</details>

**Comment by fwsmit** [comment]
Currently there is `icon_position=off` to disable icons. The problem is that it's currently not a rule, so you can only set it globally

**Comment by hokomo** [comment]
> Currently there is `icon_position=off` to disable icons. The problem is that it's currently not a rule, so you can only set it globally

Yep, exactly, and that's what I used originally, but I'd like it to be configurable on a per-notification basis. Some icons I want to hide, but others I want to preserve.

**Comment by fwsmit** [comment]
I think it's beter to make `icon_position` a rule then. I think @m-bartlett is already working on this.

**Comment by lovrolu** [comment]
I didn't even notice it has been merged! Thanks! :-)
